agents-and-function-calling/bedrock-agents/features-examples/11-get-user-confirmation-before-action-execution/11.3-create-agent-with-API-schema-and-user-confirmation/11.3-create-agent-with-API-schema-and-user-confirmation.ipynb

```diff
@@ -0,0 +1,2187 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aa89ac0d-b1af-4efa-bbb8-59a8f486a622",
+   "metadata": {},
+   "source": [
+    "Looking at the created agent, we can see its status and agent id"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 208,
+   "id": "0eef5588-1f7f-4d94-8a10-3eb8ecf316d1",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'ResponseMetadata': {'RequestId': '19bd625b-5a87-4a20-82b9-c7e5dcaf805b',\n",
+       "  'HTTPStatusCode': 201,\n",
+       "  'HTTPHeaders': {'date': 'Fri, 27 Sep 2024 17:16:48 GMT',\n",
+       "   'content-type': 'application/json',\n",
+       "   'content-length': '373',\n",
+       "   'connection': 'keep-alive',\n",
+       "   'x-amzn-requestid': '19bd625b-5a87-4a20-82b9-c7e5dcaf805b'},\n",
+       "  'RetryAttempts': 0},\n",
+       " 'Statement': '{\"Sid\":\"allow_bedrock\",\"Effect\":\"Allow\",\"Principal\":{\"Service\":\"bedrock.amazonaws.com\"},\"Action\":\"lambda:InvokeFunction\",\"Resource\":\"arn:aws:lambda:us-east-1:336118322861:function:insurance-claims-agent-us-east-1-336118322861\",\"Condition\":{\"ArnLike\":{\"AWS:SourceArn\":\"arn:aws:bedrock:us-east-1:336118322861:agent/HCUK3YNXC5\"}}}'}"
+      ]
+     },
+     "execution_count": 208,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "response"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "780ce247-ff4c-4ecf-8558-222581d2ea52",
+   "metadata": {},
+   "source": [
+    "Let's now store the agent id in a local variable to use it on the next steps"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 209,
+   "id": "4d2bf0ea-5c5c-4afc-9b4d-19158b48bb50",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "ename": "KeyError",
+     "evalue": "'agent'",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
+      "Cell \u001b[0;32mIn[209], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m agent_id \u001b[38;5;241m=\u001b[39m \u001b[43mresponse\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43magent\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124magentId\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[1;32m      2\u001b[0m agent_id\n",
+      "\u001b[0;31mKeyError\u001b[0m: 'agent'"
+     ]
+    }
+   ],
+   "source": [
+    "agent_id = response['agent']['agentId']\n",
+    "agent_id"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "32158234-b474-4bfa-8efd-afe5aa087df8",
+   "metadata": {},
+   "source": [
+    "## Create Agent Action Group\n",
+    "We will now create and agent action group that uses the lambda function and API schema files created before.\n",
```

Comment:
Can we discuss here how to set the user confirmation in an API Schema agent? Add some documentation for
`"x-requireConfirmation":"ENABLED"` and mention which functions will require the confirmation

Reply:
I had already added in session **Create S3 bucket and upload API Schema with the Required Confirmation field enabled** more details ...